### ISSUE TITLE
failed to solve: base name (${BASE_IMAGE}) should not be blank

### ISSUE THREAD

**Issue by limace255** [getsentry/self-hosted#2425]
### Self-Hosted Version

23.9.1

### CPU Architecture

x86_64

### Docker Version

24.0.6

### Docker Compose Version

2.21.0

### Steps to Reproduce

1. Pull getsentry/self-hosted repo
2. run `install.sh` script
3. run `docker compose up -d`

### Expected Result

Sentry instance should be up and running

### Actual Result

failed to solve: base name (${BASE_IMAGE}) should not be blank

**Comment by hubertdeng123** [comment]
Can you tell me what is in your .env file? `$BASE_IMAGE` should not be blank, and that is where the environment variable is being set

**Comment by limace255** [comment]
Hi @hubertdeng123 ,
Thank you for your question, I think the point is here :)
I took the .env file embedded with Self Hosted repo : 
`COMPOSE_PROJECT_NAME=sentry-self-hosted
SENTRY_EVENT_RETENTION_DAYS=90
# You can either use a port number or an IP:PORT combo for SENTRY_BIND
# See https://docs.docker.com/compose/compose-file/#ports for more
SENTRY_BIND=9000
# Set SENTRY_MAIL_HOST to a valid FQDN (host/domain name) to be able to send emails!
# SENTRY_MAIL_HOST=example.com
SENTRY_IMAGE=getsentry/sentry:nightly
SNUBA_IMAGE=getsentry/snuba:nightly
RELAY_IMAGE=getsentry/relay:nightly
SYMBOLICATOR_IMAGE=getsentry/symbolicator:nightly
VROOM_IMAGE=getsentry/vroom:nightly
WAL2JSON_VERSION=latest
HEALTHCHECK_INTERVAL=30s
HEALTHCHECK_TIMEOUT=1m30s
HEALTHCHECK_RETRIES=10`

The issue happens, I guess, for clickhouse image.
All other images are pointing on getsentry/XX:nightly, but nothing for clickhouse, and I don't know where to look for an available image.

**Comment by hubertdeng123** [comment]
Take a look into `detect-platform.sh`, that's where the clickhouse image is set

**Comment by limace255** [comment]
Hi @hubertdeng123 ,
Thank you for your feedback.
Here is what I found : 
All Shell scripts under `install` folder are not in executable mode, so `detect-platform.sh` was not executed.
If I `chmod +x install/*.sh`, it goes better for running `install.sh` script.
Then, when I `docker compose up -d`, still same issue : 
`dependency` failed to start: container sentry-self-hosted-clickhouse-1 is unhealthy`

If I check docker logs, here is the issue : 
`Processing configuration file '/etc/clickhouse-server/config.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
Merging configuration file '/etc/clickhouse-server/config.d/sentry.xml'.
Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = Exception: Failed to merge config with '/etc/clickhouse-server/config.d/sentry.xml': Access to file denied: /etc/clickhouse-server/config.d/sentry.xml, Stack trace (when copying this message, always include the lines below):

0. Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x105351b0 in /usr/bin/clickhouse
1. ? @ 0xdefbd83 in /usr/bin/clickhouse
2. DB::ConfigProcessor::loadConfig(bool) @ 0xdef9e97 in /usr/bin/clickhouse
3. BaseDaemon::reloadConfiguration() @ 0x9157010 in /usr/bin/clickhouse
4. BaseDaemon::initialize(Poco::Util::Application&) @ 0x91597d2 in /usr/bin/clickhouse
5. DB::Server::initialize(Poco::Util::Application&) @ 0x8f96458 in /usr/bin/clickhouse
6. Poco::Util::Application::run() @ 0x10457659 in /usr/bin/clickhouse
7. DB::Server::run() @ 0x8f96045 in /usr/bin/clickhouse
8. mainEntryClickHouseServer(int, char**) @ 0x8f8ce23 in /usr/bin/clickhouse
9. main @ 0x8ee8799 in /usr/bin/clickhouse
10. __libc_start_main @ 0x21b97 in /lib/x86_64-linux-gnu/libc-2.27.so
11. _start @ 0x8ee802e in /usr/bin/clickhouse
 (version 20.3.9.70 (official build))
`

**Comment by azaslavsky** [comment]
Did the now correctly permissioned install script return any errors?

In any case, the specific error you are getting indicates that you still have permission errors on at least one of your installed volumes. You may need to go through and grant the proper permissions to files like the one above - perhaps your system is configured in some way that it creates new files with maximally restrictive permissions?

**Comment by limace255** [comment]
Hi all,
Thanks again for your support, I finally managed to get Sentry deployed correctly.
@azaslavsky : you're right, the issue comes from files/folders with too restrictive permissions.
Do not know if it comes from my machine (something like a too restrictive `umask` for my Docker user), or from something else (because, here, I only download Zip Sentry Code, and unzip it then on my machine, maybe some bad rights/permissions are inherited from Source Code).

Briefly, what I did : 
`find $SENTRY_FOLDER -type f -name "*.sh"|xargs chmod a+x` --> making all Shell scripts executable, even for non-root user
`find $SENTRY_FOLDER -type d |xargs chmod a+x` --> making all folders browsable, even for non-root user
Then, launching again `install.sh` script, that worked, prompted me to Up all containers (yay !), and finally ran `docker compose up -d`

So, all issues I faced came from some scripts/folders that could not be accessed when running `install.sh` script.
